File: Dynamic Programming/acmicpc.net/1309/Main.js
const inputFilePath = "/sample.txt";
var input = require("fs")
  .readFileSync(__dirname + inputFilePath)
  // .readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split("\n");

const N = input;
const MOD = 9901;

const d = [[1, 1, 1]];

for (let i = 1; i < N; i++) {
  d.push([]);
  d[i][0] = (d[i - 1][0] + d[i - 1][1] + d[i - 1][2]) % MOD;
  d[i][1] = (d[i - 1][0] + d[i - 1][2]) % MOD;
  d[i][2] = (d[i - 1][0] + d[i - 1][1]) % MOD;
}

console.log((d[N - 1][0] + d[N - 1][1] + d[N - 1][2]) % MOD);

File: Dynamic Programming/acmicpc.net/1309/Main.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

class Main {
    static final String inputFilePath = "./Dynamic Programming/acmicpc.net/1309/sample.txt";
    static final int mod = 9901;
    // static final String inputFilePath = "./# format/sample.txt";

    public static void main(String[] args) throws Exception {
        System.setIn(new FileInputStream(inputFilePath));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int N = Integer.parseInt(br.readLine());
        int d[][] = new int[N][3];
        d[0][0] = 1;
        d[0][1] = 1;
        d[0][2] = 1;
        for (int i = 1; i < d.length; i++) {
            d[i][0] = (d[i - 1][0] + d[i - 1][1] + d[i - 1][2]) % mod;
            d[i][1] = (d[i - 1][0] + d[i - 1][2]) % mod;
            d[i][2] = (d[i - 1][0] + d[i - 1][1]) % mod;
        }
        System.out.println((d[N - 1][0] + d[N - 1][1] + d[N - 1][2]) % mod);
    }
}